ecommerce/api.py

```diff
@@ -137,10 +136,12 @@ def create_unfulfilled_order(course_id, user):
     )
     Line.objects.create(
         order=order,
-        course_key=course_run.edx_course_key,
+        course_key=course_key,
         description='Seat for {}'.format(course_run.title),
         price=price,
     )
+    if coupon is not None:
+        RedeemedCoupon.objects.create(order=order, coupon=coupon)
```

Comment:
`RedeemedCoupon` has a `UNIQUE` constraint in the database on order and coupon. What happens if the code gets to this point and there's already a row in the database for this? If it shouldn't raise an exception, you can use `get_or_create()` instead of `create()`. If it should, you should explicitly handle the exception somewhere, and have a test for it.